mkdocs/commands/build.py

```diff
@@ -324,13 +340,17 @@ def build(config, live_server=False, dump_json=False, clean_site_dir=False):
     if clean_site_dir:
         log.info("Cleaning site directory")
         utils.clean_directory(config['site_dir'])
+
     if not live_server:
         log.info("Building documentation to directory: %s", config['site_dir'])
         if not clean_site_dir and site_directory_contains_stale_files(config['site_dir']):
             log.info("The directory contains stale files. Use --clean to remove them.")
```

Comment:
Perhaps the warning should be issued here rather than in the serve command. That way it will be issued for any build. In fact, it is actually more important here, as a user could push the `site_dir` to production after running a "dirty" build. Whereas the temp site_dir of the `serve` command is always deleted when the server is stopped.